src/jackdaw/test/journal.clj

```diff
@@ -20,10 +20,8 @@
         check-condition (fn [new]
                           (when-let [result (condition? new)]
                             (remove-watch journal id)
-                            (if (boolean? result)
-                              (deliver p {:result :found})
-                              (deliver p result))))]
-
+                            (deliver p {:result :found
```

Comment:
This change addresses another usability problem encountered in the wild. Test authors could confuse the middleware if they wrote a `condition?` function that returned a map containing a `:result` key. The original intention was that the return value might be of interest to the test author when making assertions about the results but clearly it is safer to include the result in it's own `:info` key to make sure it doesn't collide with what the middleware is expecting to see.